packages/api/src/apps/api/app.ts

```diff
@@ -233,17 +249,65 @@ export default async (env: ProcessEnv) => {
 
   console.log("Starting API update loops");
 
+  // listen for new lsp contracts since after we have started api, and make sure they get state updated asap
+  // These events should only be bound after startup, since initialization above takes care of updating all contracts on startup
+  // Because there is now a event driven depdency, the lsp creator and lsp state updater must be in same process
+  serviceEvents.on("multiLspCreator", (data) => {
```

Comment:
here we listen to the event emitter, and call appropriate update functions to fetch state

Reply:
`multiLspCreator` seems like an odd name for an event. Shouldn't events generally be actions?

Reply:
I'm not sure of any conventions around this, but definately could be something like "lspFound" or something like that.  I was kind of thinking that events on a single emitter could be pathed, so maybe u could listen to them by service. A future emitter you might listen to something like `on('serviceName',(eventType,data)=>{})`, to differentiate between identical event names on different services. But this is so basic right now that im happy to change it to whatever you think is best

Reply:
Ahhhh, that makes sense! I think we should either do service name + event type or a global event name (some sort of action). Service name alone is confusing IMO since it doesn't imply _what_ is happening with the service.

Reply:
ive added the event type, so the service can specify the type of event that happened.  Currently its just named "created" to signify that a contract was created within the service.